src/psyclone/tests/nemo/nemo_config_test.py

```diff
@@ -0,0 +1,121 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author J. Henrichs, Bureau of Meteorology
+
+''' Module containing tests for nemo specific config files.'''
+
+from __future__ import absolute_import
+
+import pytest
+
+from psyclone.configuration import Config, ConfigurationError
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup():
+    '''Make sure that all tests here use nemo API, and that to clean
+    the config file at the end of the tests.'''
+    Config.get().api = "nemo"
+    yield()
+    # At the end of all tests make sure that we wipe the Config object
+    # so we get a fresh/default one for any further test (and not a
+    # left-over one from a test here).
+    Config._instance = None
+
+
+# =============================================================================
+def test_invalid_nemo_config_files(tmpdir):
+    ''' Test various error conditions.
+    '''
+
+    # Valid configuration file without nemo-specific settings.
+    # We add several lines in the tests for various error conditions
+    # pylint: disable=invalid-name
+    _CONFIG_CONTENT = '''\
+    [DEFAULT]
+    DEFAULTAPI = nemo
+    DEFAULTSTUBAPI = dynamo0.3
+    DISTRIBUTED_MEMORY = true
+    REPRODUCIBLE_REDUCTIONS = false
+    REPROD_PAD_SIZE = 8
+    [nemo]
+    '''
+    # Create a config files with a nemo section and a mapping-lon
+    # but leave out each required key
+    for (key, data) in [("var", "start: 1, stop: jpi"),
+                        ("start", "var: ji, stop: jpi"),
+                        ("stop", "var: ji, start: 1")]:
+        content = _CONFIG_CONTENT + "mapping-lon = " + data
+        Config._instance = None
+        config_file = tmpdir.join("config1")
+        with config_file.open(mode="w") as new_cfg:
+            new_cfg.write(content)
+            new_cfg.close()
+
+            config = Config()
+            with pytest.raises(ConfigurationError) as err:
+                config.load(str(config_file))
+            assert "does not contain key '{0}".format(key) in str(err)
+
+    # Add an invalid index-order
+    content = _CONFIG_CONTENT + \
+        '''mapping-lon = var: i, start: 1, stop:2
+    mapping-lat = var: i, start: 1, stop:2
```

Comment:
You've used the same variable name twice here ("i") which should cause an error. Please add code for this plus a test.

Reply:
Done (but I also changed this code to use correct/not-duplicated names.

Reply:
Great!